service/dap/server.go

```diff
@@ -450,7 +464,9 @@ func (s *Server) handleRequest(request dap.Message) {
 	// the next stop. In addition, the editor itself might block waiting
 	// for these requests to return. We are not aware of any requests
 	// that would benefit from this approach at this time.
-	if s.debugger != nil && s.debugger.IsRunning() {
+	if s.debugger != nil && s.isRunning() {
+		s.resumeMu.Lock()
```

Comment:
This is so that another goroutine cannot resume execution after it has requested the program to halt. If another goroutine were to resume execution while breakpoints were being set, this could end up with a weird state where some breakpoints are set and others are not.

Would a better name be stateChangeMu? 

I had moved this to only apply to setBreakpoints, but I think it also will need to be held for pause, so that the pause 
 will definitely be seen by run().